src/Calculator/Views/CalculatorProgrammerBitFlipPanel.xaml.cs

```diff
@@ -0,0 +1,288 @@
+            {
+                int lastPosition = -1;
+                if (Model != null)
+                {
+                    lastPosition = GetIndexOfLastBit(Model.ValueBitLength);
+                }
+
+                if (position == lastPosition)
+                {
+                    bitPosition = resourceLoader.GetResourceString("MostSignificantBit");
+                }
+                else
+                {
+                    string indexName = resourceLoader.GetResourceString(position.ToString());
+                    string bitPositionTemplate = resourceLoader.GetResourceString("BitPosition");
+                    bitPosition = LocalizationStringUtil.GetLocalizedString(bitPositionTemplate, indexName);
+                }
+            }
+            return LocalizationStringUtil.GetLocalizedString(automationNameTemplate, bitPosition, value ? "1" : "0");
+        }
+
+        private void UpdateCheckedStates(bool updateAutomationPropertiesNames)
+        {
+            Debug.Assert(!m_updatingCheckedStates);
+            Debug.Assert(m_flipButtons.Length == s_numBits);
+
+            if (Model == null)
+            {
+                return;
+            }
+
+            m_updatingCheckedStates = true;
+            // CSHARP_MIGRATION:
+            // iterator and index move at the same time with same step
+            // add index validation in the loop
+            int index = 0;
+            bool mustUpdateTextOfMostSignificantDigits = m_currentValueBitLength != Model.ValueBitLength;
+            int previousMSDPosition = GetIndexOfLastBit(m_currentValueBitLength);
+            int newMSDPosition = GetIndexOfLastBit(Model.ValueBitLength);
+            foreach (bool val in Model.BinaryDigits)
+            {
+                if (index < m_flipButtons.Length)
+                {
+                    bool hasValueChanged = m_flipButtons[index].IsChecked.Value != val;
+                    m_flipButtons[index].IsChecked = val;
+                    if (updateAutomationPropertiesNames
+                        || hasValueChanged
+                        || (mustUpdateTextOfMostSignificantDigits && (index == previousMSDPosition || index == newMSDPosition)))
+                    {
+                        m_flipButtons[index].SetValue(AutomationProperties.NameProperty, GenerateAutomationPropertiesName(index, val));
+                    }
+                    ++index;
+                }
+            }
+
+            m_updatingCheckedStates = false;
+        }
+
+        private void UpdateAutomationPropertiesNames()
+        {
+            foreach (FlipButtons flipButton in m_flipButtons)
+            {
+                int index = (int)flipButton.Tag;
+                flipButton.SetValue(AutomationProperties.NameProperty, GenerateAutomationPropertiesName(index, flipButton.IsChecked.Value));
+            }
+        }
+
+        // CSHARP_MIGRATION: TODO:
+        // c++ const method. c# readonly method is available for c# 8.0+
```

Comment:
Since we are treating all the const method of C++/CX as non-const method in C#,  we can remove this comment here.

Reply:
Removed